domain-models-runtime/src/test/java/org/folio/rest/persist/PgUtilIT.java

```diff
@@ -1107,6 +1107,49 @@ public void optimizedSQLwithNo500(TestContext testContext) {
         }));
   }
 
+   @Test
+  public void deleteByCQLwithNo500(TestContext testContext) {
+    PostgresClient pg = PostgresClient.getInstance(vertx, "testtenant");
+    insert(testContext, pg, "delete_test",  1);
+    PgUtil.delete("users",  "username=delete_test",
+        okapiHeaders, vertx.getOrCreateContext(), ResponseWithout500.class, testContext.asyncAssertFailure(e -> {
+          assertThat(e, is(instanceOf(NullPointerException.class)));
+        }));
+  }
+
+  @Test
+  public void deleteByCQLwithNo204(TestContext testContext) {
+    String uuid = randomUuid();
+    post(testContext, "Pippilotta", uuid, 201);
+
+    PgUtil.delete("users", "username=delete_test",
+        okapiHeaders, vertx.getOrCreateContext(),
+        Users.PutUsersByUserIdResponse.class,
+        asyncAssertSuccess(testContext, 204));
+  }
+
+  @Test
+  public void deleteByCQLwithNo400(TestContext testContext) {
+    PostgresClient pg = PostgresClient.getInstance(vertx, "testtenant");
+    setUpUserDBForTest(testContext, pg);
+    exception.expect(NullPointerException.class);
+    UserdataCollection userdataCollection = new UserdataCollection();
+    Async async = testContext.async();
+    PgUtil.delete("users","username=b", okapiHeaders,
+        vertx.getOrCreateContext(), ResponseWithout400.class, testContext.asyncAssertSuccess(response -> {
+          if (response.getStatus() != 500) {
+            testContext.fail("Expected status 500, got "
+                + response.getStatus() + " " + response.getStatusInfo().getReasonPhrase());
+            async.complete();
+            return;
+          }
+
+          async.complete();
+    }));
+    async.awaitSuccess(10000 /* ms */);
+  }
```

Comment:
Please remove all unnecessary test setup. See deleteByIdResponseWithout500 and deleteByIdResponseWithout204 for examples.